### DISCUSSION TITLE
'NoneType' object has no attribute 'write'

### DISCUSSION THREAD

**Discussion by timothyreed1** [easlice/bandcamp-downloader#47] [Q&A]
Hi -  installed the latest Bandcamp-downloader on my Mac running the latest OS and am getting the error: "AttributeError: 'NoneType' object has no attribute 'write'". I've tried logging into Bandcamp on Brave and Chrome with the same error. I have about 210 purchases on Bandcamp and can download manually. Has anyone run into this issue before? 

Thanks to easlice for making this software available.

% bandcamp-downloader.py --browser brave _my bandcamp username_
Traceback (most recent call last):
  File "/Users/me/.local/bin/bandcamp-downloader.py", line 591, in <module>
    sys.exit(main())
             ~~~~^^
  File "/Users/me/.local/bin/bandcamp-downloader.py", line 186, in main
    items = get_items_for_user(args.username, args.include_hidden)
  File "/Users/me/.local/bin/bandcamp-downloader.py", line 289, in get_items_for_user
    items.update(fetch_items(
                 ~~~~~~~~~~~^
        False,
        ^^^^^^
    ...<2 lines>...
        # count is the number we have left to fetch after the initial data blob
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        data['collection_data']['item_count'] - len(data['item_cache']['collection'])))
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/me/.local/bin/bandcamp-downloader.py", line 248, in fetch_items
    return merge_items_and_urls(data['items'], data['redownload_urls'] or {})
  File "/Users/me/.local/bin/bandcamp-downloader.py", line 337, in merge_items_and_urls
    CONFIG['TQDM'].write("WARN: couldn't find redownload URL for item_id:[{}], artist:[{}], title:[{}]".format(
    ^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'write'

**Comment by easlice** [comment]
Hey @timothyreed1, sorry for taking so long to get back.

That error is technically because the TQDM item (the thing that reports the progress of the script) wasn't there... which is odd, to say the least...
Ah, ok, I see what is going on. It was supposed to tell you it couldn't find a download url for an item, but this happens before we're doing 'album processing', so the TQDM object hasn't been created yet.

I'll get that fixed.

**Reply by easlice** [reply]
@timothyreed1 This should get fixed by this #49. Once the tests run through I'll merge it. If you would, you could try again with the new version and let me know if your issue is resolved?

**Reply by reagan9000** [reply]
Thank you!

**Comment by easlice** [comment]
Closing this, as it should be fixed. If it isn't feel free to reopen it.
